test/spec/grammar/ast_builder_test.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable key-spacing, no-unused-expressions */
+const expect = require('chai').expect;
+const parse = require('../../../lib/dsl/poc/api').parse;
+const buildAst = require('../../../lib/dsl/poc/ast_builder').buildAst;
+
+describe('ASTBuilder', () => {
+  it('can build an ast for a simple entity', () => {
+    const input = 'entity Person { name string }';
+    const parseResult = parse(input);
+    expect(parseResult.lexErrors).to.be.empty;
+    expect(parseResult.parseErrors).to.be.empty;
+
+    const ast = buildAst(parseResult.cst);
+    expect(ast).to.deep.equal({
+      constants: [],
+      entities:  [
+        {
+          fields:    [
+            {
+              name:        'name',
+              type:        'string',
+              validations: []
+            }
+          ],
+          name:      'Person',
+          tableName: undefined
+        }
+      ]
+    });
+  });
+
+  it('can build an ast for a complex JDL', () => {
+    const input = `
+    entity Person (HRXD1)
+    { 
+      name string, 
+      age number max(max_age)
+    }
+    
+    max_age = 120
```

Comment:
Constants should be uppercase

Reply:
That is one of the quirks I have not documented yet.
You cannot have different kinds of identifiers syntax. At least not if you can not always distinguish
between them easily. 

So what happens is that we will have a single basic identifier (like Java/ECMAScript/most languages) and the validation would be performed POST parsing.
e.g like a linting rule "constants should be upper case".

This means the Chevrotain syntax would be a **superset** (hopefully **strict** superset).
Of the grammar and the additional validations would ensure the input is not only **syntactically** valid but also **semantically** valid.

This is again what happens commonly in compilers (lexing -> parsing -> ast -> resolving -> validating -> code generation)